cache/src/main/java/io/envoyproxy/controlplane/cache/SimpleCache.java

```diff
@@ -145,7 +145,12 @@ public Watch createWatch(
       }
 
       // Otherwise, the watch may be responded immediately
-      respond(watch, snapshot, group);
+      boolean responded = respond(watch, snapshot, group);
+
+      if (!responded) {
+        watchCount++;
+        status.setWatch(watchCount, watch);
```

Comment:
We're leaving a new open watch here, so I think this section will need to look more like the open watch block above, i.e.:

```java
if (!responded) {
  watchCount++;

  long watchId = watchCount;

  LOGGER.info("did not respond immediately, leaving open watch {} for {}[{}] from node {} for version {}",
      watchId,
      request.getTypeUrl(),
      String.join(", ", request.getResourceNamesList()),
      group,
      request.getVersionInfo());

  status.setWatch(watchId, watch);

  watch.setStop(() -> status.removeWatch(watchId));
}
```

It would also be good to create a test in `SimpleCacheTest` (similar to https://github.com/envoyproxy/java-control-plane/blob/450d0b3cdcfa336fb5c79e5178913df6e7d76bde/cache/src/test/java/io/envoyproxy/controlplane/cache/SimpleCacheTest.java#L132) that exercises the new flow.

Reply:
`watchIsLeftOpenIfNotRespondedImmediately` is added to test the new flow.